Modules/zlibmodule.c

```diff
@@ -1371,61 +1372,61 @@ PyDoc_STRVAR(zlib_module_documentation,
 "objects support decompress() and flush().");
 
 static int
-zlib_clear(PyObject *m)
+zlib_clear(PyObject *mod)
 {
-    zlibstate *state = get_zlib_state(m);
+    zlibstate *state = get_zlib_state(mod);
     Py_CLEAR(state->Comptype);
     Py_CLEAR(state->Decomptype);
     Py_CLEAR(state->ZlibError);
     return 0;
 }
 
 static int
-zlib_traverse(PyObject *m, visitproc visit, void *arg)
+zlib_traverse(PyObject *mod, visitproc visit, void *arg)
 {
-    zlibstate *state = get_zlib_state(m);
+    zlibstate *state = get_zlib_state(mod);
     Py_VISIT(state->Comptype);
     Py_VISIT(state->Decomptype);
     Py_VISIT(state->ZlibError);
     return 0;
 }
 
 static void
-zlib_free(void *m)
+zlib_free(void *mod)
 {
-    zlib_clear((PyObject *)m);
+    zlib_clear((PyObject *)mod);
 }
 
 static int
-zlib_exec(PyObject *m)
+zlib_exec(PyObject *mod)
 {
-    zlibstate *state = get_zlib_state(m);
+    zlibstate *state = get_zlib_state(mod);
 
     state->Comptype = (PyTypeObject *)PyType_FromModuleAndSpec(
-        m, &Comptype_spec, NULL);
+        mod, &Comptype_spec, NULL);
     if (state->Comptype == NULL) {
         return -1;
     }
 
     state->Decomptype = (PyTypeObject *)PyType_FromModuleAndSpec(
-        m, &Decomptype_spec, NULL);
+        mod, &Decomptype_spec, NULL);
     if (state->Decomptype == NULL) {
         return -1;
     }
 
     PyObject *ZlibError = PyErr_NewException("zlib.error", NULL, NULL);
     if (ZlibError != NULL) {
```

Comment:
I prefer judging the `ZlibError== NULL`  directly. something like:
```
if (ZlibError == NULL) {
    Py_DECREF(state->Comptype);
    Py_DECREF(state->Decomptype);
    return -1;
}
```
I don't know what will happen when some objects is NULL. Maybe you can update it in your PR, depend on you ;)

Reply:
I don't see why you only want to clear two state attributes. If the exec function fails, the module will be destroyed, and before that its free function will be called: this function clears all state attributes. There is no need to clear it manually.

```
    Py_DECREF(state->Comptype);
    Py_DECREF(state->Decomptype);
```

is likely to crash since you don't clear the attribute.

In short, `return -1` is correct.

Reply:
But I agree that the exec function must fail if PyErr_NewException fails. I suggest something like:

```
state->ZlibError =  PyErr_NewException("zlib.error", NULL, NULL);
if (state->ZlibError == NULL) {
    return -1;
}
// ... code calling PyModule_AddObject ...
```

Reply:
> I don't see why you only want to clear two state attributes. If the exec function fails, the module will be destroyed
Oh, sorry. You are right. I forget this detail~